core/src/main/java/org/infinispan/marshall/core/BytesObjectInput.java

```diff
@@ -0,0 +1,250 @@
+package org.infinispan.marshall.core;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+
+final class BytesObjectInput implements ObjectInput, PositionalBuffer.Input {
+
+   final byte bytes[];
+
+   final GlobalMarshaller marshaller;
+
+   int pos;
+   int offset; // needed for external JBoss Marshalling, to be able to rewind correctly when the bytes are prepended :(
```

Comment:
This class is package default and it's `final`, so `protected` adds nothing here. If anything `private` but I didn't feel the need for it.